Mortgage-loan-calculator/src/app/admin-panel/admin-panel.component.ts

```diff
@@ -1,42 +0,0 @@
-import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
-import { MatPaginator } from '@angular/material/paginator';
-import { MatTableDataSource } from '@angular/material/table';
-import { CustomerService } from './services/customer.service';
-import { Customer } from '../types';
-import { CalculatorFormComponent } from '../calculator-form/calculator-form.component';
-import {StorageService} from "../admin-login/services/storage.service";
-
-@Component({
-  selector: 'app-admin-panel',
-  templateUrl: './admin-panel.component.html',
-  styleUrls: ['./admin-panel.component.css'],
-})
-export class AdminPanelComponent implements AfterViewInit, OnInit {
-  constructor (private service: CustomerService) {
-  }
-  ngOnInit(){
-
-  }
-
-  customers: MatTableDataSource<Customer> = new MatTableDataSource<Customer>();
-
-  displayedColumns: string[] = [
-    'name',
-    'phoneNumber',
-    'email',
-    'ip',
-    'time',
-    'action',
-  ];
-
-  @ViewChild(MatPaginator, { static: true }) paginator!: MatPaginator;
-  @ViewChild(CalculatorFormComponent)
-  calculatorFormComponent?: CalculatorFormComponent;
-  ngAfterViewInit() {
-    this.service.getCustomer().subscribe((data) => {
-      this.customers.data = data;
```

Comment:
Avoid data extraction from  Observables. Instead us Observable value with `async` pipe in  template directly.